lib/medusa/broker.ex

```diff
@@ -45,17 +50,41 @@ defmodule Medusa.Broker do
 
   def handle_cast({:publish, event, payload}, state) do
     Logger.debug "#{inspect __MODULE__}: [#{inspect event}]: #{inspect payload}"
-    Enum.each(state, fn(e) ->
-      if Regex.match?(e, event) do
-        @adapter.insert(base64_encode_regex(e), payload)
-        GenServer.cast (base64_encode_regex(e) |> String.to_atom), {:trigger}
-      end
-    end)
+    Enum.each(state, &maybe_route(&1, event, payload))
```

Comment:
I think we should be able to trigger these using `Task.async` or even `Task.Supervisor.async` for that matter